config/initializers/rack_attack.rb

```diff
@@ -1,15 +1,32 @@
 require 'http_authentication_token'
 
+class Rack::Attack::Request
+  extend Mandate::Memoize
+
+  def auth_token
+    HttpAuthenticationToken.from_header(env['HTTP_AUTHORIZATION'])
+  end
+
+  memoize
+  def route_name
+    route_name = nil
+    Rails.application.routes.router.recognize(self) do |route|
+      route_name = route.name if route.name.present?
+    end
+    route_name
+  end
+end
```

Comment:
Now that I had two throttle blocks, I decided that it might be nice to extract this shared logic and monkey patch it.